src/java/fr/paris/lutece/plugins/oauth2/dataclient/AbstractDataClient.java

```diff
@@ -250,6 +263,12 @@ public String getData( Token token )
         {
             RequestAuthenticator authenticator = new BearerTokenAuthenticator( token.getAccessToken( ) );
             strResponse = httpAccess.doGet( strUrl, authenticator, null );
+            
+            if ( StringUtils.isNotEmpty( strResponse ) &&_jWTParser != null && _bEnableJwtParser )
+            {                      
+                strResponse = parseJWTResponse( strResponse );
```

Comment:
Why not make another class for dataclients whose data is a JWT ? You would have DataClient -> AbstractDataClient -> AbstractJWTDataClient